                                  #Exercice 1 : Introduction à l'analyse des données (Facile)
                                  
1.L'analyse de données est la science et le processus d'examen, de nettoyage, de transformation et de modélisation des données brutes afin de découvrir des modèles cachés, d'extraire des informations exploitables et de guider la prise de décision stratégique.

2.L'importance de l'analyse de données dans les contextes modernes: 
Remplacement de l’intuition par les faits
Prise de décision en temps réel
Optimisation opérationnelle
Démocratisation technologique

3.Trois domaines d'application actuels et concrets:
Le Marketing et la Personnalisation de l'Expérience Client
La Santé et la Médecine Prédictive
La Logistique et la Maintenance Prédictive industrielle

Exemple concret : Les géants de la livraison comme Uber Eats analysent en direct les conditions météorologiques, le trafic routier et le temps de préparation moyen des restaurants pour optimiser les tournées des livreurs. Dans les usines, cette approche permet de planifier les réparations de machines uniquement lorsque les algorithmes détectent une anomalie de vibration, évitant ainsi des arrêts de production extrêmement coûteux.

In [34]:
                                  #Exercice 2 : Chargement et analyse initiale des données
import pandas as pd 
clean_dataset = pd.read_csv("clean_dataset.csv")
clean_dataset.head()

# Ce programme donne les 5 premiers elements de chaque fichier 


,Gender,Age,Debt,Married,BankCustomer,Industry,Ethnicity,YearsEmployed,PriorDefault,Employed,CreditScore,DriversLicense,Citizen,ZipCode,Income,Approved
0,1,30.83,0.000,1,1,Industrials,White,1.25,1,1,1,0,ByBirth,202,0,1
1,0,58.67,4.460,1,1,Materials,Black,3.04,1,1,6,0,ByBirth,43,560,1
2,0,24.50,0.500,1,1,Materials,Black,1.50,1,0,0,0,ByBirth,280,824,1
3,1,27.83,1.540,1,1,Industrials,White,3.75,1,1,5,1,ByBirth,100,3,1
4,1,20.17,5.625,1,1,Industrials,White,1.71,1,0,0,0,ByOtherMeans,120,0,1


In [51]:
Crx=pd.read_csv("crx.csv")
Crx.head()

,b,30.83,0,u,g,w,v,1.25,t,t.1,01,f,g.1,00202,0.1,+
0,a,58.67,4.460,u,g,q,h,3.04,t,t,6,f,g,00043,560,+
1,a,24.50,0.500,u,g,q,h,1.50,t,f,0,f,g,00280,824,+
2,b,27.83,1.540,u,g,w,v,3.75,t,t,5,t,g,00100,3,+
3,b,20.17,5.625,u,g,w,v,1.71,t,f,0,f,s,00120,0,+
4,b,32.08,4.000,u,g,m,v,2.50,t,f,0,t,g,00360,0,+


In [35]:
Time_americains=pd.read_csv("Time_Americans_Spend_Sleeping.csv")
Time_americains.head()


,index,Year,Period,Avg hrs per day sleeping,Standard Error,Type of Days,Age Group,Activity,Sex
0,0,2003,Annual,8.57,0.018,All days,15 years and over,Sleeping,Both
1,1,2004,Annual,8.55,0.026,All days,15 years and over,Sleeping,Both
2,2,2005,Annual,8.62,0.023,All days,15 years and over,Sleeping,Both
3,3,2006,Annual,8.63,0.024,All days,15 years and over,Sleeping,Both
4,4,2007,Annual,8.57,0.024,All days,15 years and over,Sleeping,Both


In [36]:
Mental_heath=pd.read_csv("Mental_health_Depression_disorder_Data.csv", low_memory=False)
Mental_heath.head()

,index,Entity,Code,Year,Schizophrenia (%),Bipolar disorder (%),Eating disorders (%),Anxiety disorders (%),Drug use disorders (%),Depression (%),Alcohol use disorders (%)
0,0,Afghanistan,AFG,1990,0.16056,0.697779,0.101855,4.828830,1.677082,4.071831,0.672404
1,1,Afghanistan,AFG,1991,0.160312,0.697961,0.099313,4.829740,1.684746,4.079531,0.671768
2,2,Afghanistan,AFG,1992,0.160135,0.698107,0.096692,4.831108,1.694334,4.088358,0.670644
3,3,Afghanistan,AFG,1993,0.160037,0.698257,0.094336,4.830864,1.705320,4.096190,0.669738
4,4,Afghanistan,AFG,1994,0.160022,0.698469,0.092439,4.829423,1.716069,4.099582,0.669260


                            #Exercice 3 : Identification des types de données

Ensemble de données 1 : Troubles de la santé mentale
 Colonnes qualitativesEntity : Le nom du pays ou de la région (ex. Afghanistan). C'est une catégorie textuelle nominale.Code : Le code à trois lettres du pays (ex. AFG). C'est un identifiant textuel abrégé.index : Un identifiant numérique de ligne. C'est une variable qualitative nominale. Faire la moyenne des index n'a aucun sens.

Colonnes quantitativesYear : L'année de collecte. C'est une variable quantitative d'intervalle. Elle permet de mesurer l'écoulement du temps.Les colonnes de troubles (Schizophrenia (%), Bipolar disorder (%), etc.) : Le pourcentage de la population touché. Ce sont des variables quantitatives continues. Elles mesurent une proportion précise.

Ensemble de données 2 : Sommeil des Américains
Colonnes qualitativesPeriod : La période de l'étude (ex. Annual). C'est une étiquette textuelle.Type of Days : Le type de jour analysé (ex. All days). C'est une catégorie textuelle.Age Group : La tranche d'âge ciblée (ex. 15 years and over). C'est une variable qualitative ordinale (les tranches d'âge suivent une logique d'ordre).Activity : L'activité mesurée (ex. Sleeping). C'est une variable textuelle nominale.Sex : Le genre des individus (ex. Both). C'est une variable catégorielle nominale.index : Identifiant de ligne (qualitatif nominal, identique au premier dataset).Colonnes quantitativesYear : L'année de l'observation (quantitative d'intervalle).Avg hrs per day sleeping : Le nombre moyen d'heures de sommeil. C'est une variable quantitative continue. Elle représente une durée mesurable.Standard Error : L'erreur type de la mesure. C'est une variable quantitative continue représentant un écart statistique numérique.

Ensemble de données 3 : Demandes de carte de créditNote importante : Cet ensemble contient des données qualitatives pré-encodées sous forme de chiffres (0 ou 1).Colonnes qualitativesGender : Le genre, codé par 0 ou 1. C'est une variable qualitative nominale (binaire). Le chiffre est une simple étiquette.Married : Le statut marital, codé par des chiffres (ex. 1). C'est une variable qualitative nominale.BankCustomer : Le fait d'être client de la banque ou non, codé numériquement. C'est une variable qualitative nominale.Industry : Le secteur d'activité (ex. Industrials). C'est une catégorie textuelle nominale.Ethnicity : L'origine ethnique (ex. White). C'est une catégorie textuelle nominale.PriorDefault : L'historique de défaut de paiement, codé par 0 ou 1. C'est une variable qualitative nominale (binaire).Employed : Le statut d'emploi (actif ou non), codé par 0 ou 1. C'est une variable qualitative nominale (binaire).DriversLicense : La possession d'un permis de conduire, codé par 0 ou 1. C'est une variable qualitative nominale (binaire).Citizen : Le statut de citoyenneté (ex. ByBirth). C'est une catégorie textuelle nominale.ZipCode : Le code postal (ex. 202). C'est une variable qualitative nominale. Additionner deux codes postaux n'a aucune signification mathématique.Approved : Décision d'approbation du crédit, codée par 0 ou 1. C'est une variable qualitative nominale (binaire).

Colonnes quantitativesAge : L'âge du demandeur en années. C'est une variable quantitative continue ou discrète (mesure numérique d'une durée).Debt : Le montant de la dette. C'est une variable quantitative continue représentant une valeur monétaire.YearsEmployed : Le nombre d'années passées chez l'employeur actuel. C'est une variable quantitative continue.CreditScore : Le score de crédit (ex. 6). C'est une variable quantitative discrète qui mesure mathématiquement la solvabilité.Income : Les revenus du demandeur. C'est une variable quantitative continue (valeur financière).

 Colonnes qualitatives (Catégorielles)Colonne 1 (b, a, a...) : Représente le genre (Gender). C'est une variable qualitative nominale (binaire).Colonne 4 (u...) : Représente le statut marital (Married). Variable qualitative nominale.Colonne 5 (g...) : Représente le type de client bancaire (BankCustomer). Variable qualitative nominale.Colonne 6 (w, q, m...) : Représente le secteur d'activité (Industry). Variable qualitative textuelle nominale.Colonne 7 (v, h...) : Représente l'origine ethnique (Ethnicity). Variable qualitative textuelle nominale.Colonne 9 (t...) : Historique de défaut de paiement (PriorDefault). Code qualitatif binaire (t pour True / f pour False).Colonne 10 (t.1, t, f...) : Statut d'emploi (Employed). Code qualitatif binaire (t/f).Colonne 12 (f, t...) : Possession du permis de conduire (DriversLicense). Code qualitatif binaire (t/f).Colonne 13 (g.1, g, s...) : Statut de citoyenneté (Citizen). Variable qualitative nominale.Colonne 14 (00202, 00043...) : Code postal (ZipCode). Bien qu'écrit avec des chiffres, c'est un identifiant géographique qualitatif.Colonne 16 (+) : Statut d'approbation du crédit (Approved). Variable qualitative binaire où + signifie approuvé et - signifie refusé.
 
  Colonnes quantitatives (Numériques)Colonne 2 (30.83, 58.67...) : L'âge du demandeur (Age). Variable quantitative continue.Colonne 3 (0, 4.460...) : Le montant de la dette (Debt). Variable quantitative continue.Colonne 8 (1.25, 3.04...) : Le nombre d'années chez l'employeur (YearsEmployed). Variable quantitative continue.Colonne 11 (0, 6, 5...) : Le score de crédit (CreditScore). Variable quantitative discrète (valeurs entières).Colonne 15 (0.1, 560, 824...) : Le niveau de revenus (Income). Variable quantitative continue.

In [ ]:
                               #Exercice 4 : Exploration des types de données
iris = pd.read_csv("Iris.csv")
iris.head()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species
0,1,5.1,3.5,1.4,0.2,Iris-setosa
1,2,4.9,3.0,1.4,0.2,Iris-setosa
2,3,4.7,3.2,1.3,0.2,Iris-setosa
3,4,4.6,3.1,1.5,0.2,Iris-setosa
4,5,5.0,3.6,1.4,0.2,Iris-setosa


1. les colonnes qualitatives: Species
2. les colonnes quantitatives: SepalLenghCm,SepalWidthCm,PetalLengthCm,PetalWidthCm
3. la Specie est qualitative parce qu'elle comporte des valeurs descriptive 
Les Sepal quant a elle comporte des valeurs numerique 

In [ ]:
                                #Exercice 5 : Compétences de base en observation et analyse de données
Time = pd.read_csv("Time Americans Spend Sleeping.csv")
Time.head()

,index,Year,Period,Avg hrs per day sleeping,Standard Error,Type of Days,Age Group,Activity,Sex
0,0,2003,Annual,8.57,0.018,All days,15 years and over,Sleeping,Both
1,1,2004,Annual,8.55,0.026,All days,15 years and over,Sleeping,Both
2,2,2005,Annual,8.62,0.023,All days,15 years and over,Sleeping,Both
3,3,2006,Annual,8.63,0.024,All days,15 years and over,Sleeping,Both
4,4,2007,Annual,8.57,0.024,All days,15 years and over,Sleeping,Both


1. A. Analyse de tendances temporelles Objectif : Comprendre comment le temps de sommeil des Américains évolue au fil des ans.Colonnes à retenir : Year et Avg hrs per day sleeping.Justification : La colonne Year sert d'axe temporel (variable indépendante) et Avg hrs per day sleeping mesure la variable d'intérêt (variable dépendante). En option, Standard Error permet d'ajouter des barres d'erreur pour évaluer la précision des estimations annuelles.
B. Analyse comparative (Segmentation démographique)Objectif : Identifier quels groupes de population dorment le plus ou le moins.Colonnes à retenir : Sex, Age Group et Avg hrs per day sleeping.Justification : Sex et Age Group sont des variables catégorielles. Elles permettent de segmenter la population afin de comparer les moyennes de sommeil entre les hommes et les femmes, ou entre les différentes tranches d'âge.
C. Analyse contextuelle (Impact du mode de vie)Objectif : Déterminer si le rythme de la semaine influence le sommeil.Colonnes à retenir : Type of Days et Avg hrs per day sleeping.Justification : La colonne Type of Days (qui distingue généralement les jours de semaine des week-ends/jours fériés) permet d'isoler l'impact des obligations professionnelles ou scolaires sur le repos.

                                        #Exercice 6 : Identification des types de données
1. Données structurées
  Les rapports financiers d'une entreprise dans un fichier Excel : Les données y sont organisées de manière ordonnée en lignes et en colonnes, avec des types bien définis (chiffres, dates, textes) facilitant le calcul

   Données d'inventaire dans une base de données relationnelle : Il s'agit de l'exemple type de données structurées. Elles suivent un schéma strict (tables SQL) avec des relations bien définies entre les éléments (ex: ID produit, quantité)

2.Données non structurées
 Photographies téléchargées sur une plateforme de médias sociaux : Les fichiers d'images (JPEG, PNG) sont des médias riches qui ne possèdent pas de format textuel ou relationnel sous-jacent.

Une collection d'articles de presse sur un site web : Même si le site web a une mise en page ordonnée, le cœur de la donnée est du texte libre (langage naturel), qui est par nature non structuré.

Enregistrements d'entretiens issus d'une étude de marché : Qu'il s'agisse de fichiers audio, vidéo ou de transcriptions textuelles mot à mot, le contenu varie librement d'un entretien à un autre sans format fixe

                                        #Exercice 7 : Exercice de transformation

1.Articles de blog sur les expériences de voyageMéthode : Le Traitement du Langage Naturel (NLP) couplé à la Reconnaissance d'Entités Nommées (NER). On utilise un Large Language Model (LLM) pour extraire des champs précis et les enregistrer dans un fichier au format JSON ou CSV.Champs structurés obtenus : Nom de la destination, durée du séjour, budget estimé, activités mentionnées, note de sentiment (positif/négatif).Justification : Le texte des blogs est rédigé en langage naturel varié. Le NLP est l'unique approche capable de comprendre le contexte linguistique afin d'isoler des entités géographiques ou sémantiques précises sans nécessiter un format d'écriture uniforme.

 2.Enregistrements audio des appels clientMéthode : Une pipeline combinant la Transcription Speech-to-Text (STT) suivie d'une Analyse de Sentiment automatique. Le logiciel convertit la voix en texte brut, puis un algorithme catégorise l'appel.Champs structurés obtenus : ID de l'appelant, motif de l'appel (technique, facturation, résiliation), durée de l'appel, score de satisfaction client, mots-clés récurrents.Justification : Les fichiers audio ne sont pas exploitables directement par des bases de données relationnelles. La transcription textuelle intermédiaire est indispensable avant d'appliquer des filtres de classification quantitatifs.

3.Notes manuscrites de brainstormingMéthode : La Reconnaissance Optique de Caractères (OCR) intelligente (ou ICR, spécialisée dans l'écriture manuscrite). Les blocs de texte numérisés sont ensuite soumis à un algorithme de clustering (regroupement de données) pour catégoriser les idées.Champs structurés obtenus : Date de la session, nom du contributeur, catégorie de l'idée (marketing, produit, RH), liste des actions à mener.Justification : L'OCR transforme les pixels d'une image ou d'un scan en chaînes de caractères lisibles par une machine. Sans cette conversion de base, le texte reste prisonnier du format image et est impossible à indexer.

4.Tutoriel vidéo sur la cuisineMéthode : L'Analyse Multimodale par IA. Elle combine la vision par ordinateur (pour reconnaître les actions et les ingrédients à l'écran), l'extraction des métadonnées vidéo et le traitement de la piste audio (transcription de la voix du chef).Champs structurés obtenus : Titre de la recette, liste des ingrédients, quantités détectées, temps de cuisson total, étapes séquentielles (étape 1, étape 2, etc.).Justification : Une vidéo superpose des flux visuels et auditifs. L'analyse multimodale permet de synchroniser ce que le chef dit (piste audio) avec ce qu'il fait (flux vidéo), garantissant une extraction fiable de la chronologie de la recette.
                    

In [ ]:
                                       #Exercice 8 : Importer un fichier depuis Kaggle
train = pd.read_csv("train.csv")
train.head()


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [ ]:
                                       # Exercice 9 : Exporter un dataframe au format Excel et au format JSON
Data_frame = {
    "moyenne": [12, 13, 13, 15, 12],
    "age": [16, 17, 15, 18, 15],
    "nom": ["Dessousa", "Albert", "Gilbert", "Sery", "Levi"]
}
df = pd.DataFrame(Data_frame)

df.to_csv('Data_frame.csv')
df.to_json('Data_frame.json')


In [ ]:
                                               #Exercice 10 : Lecture de données JSON
Posts = pd.read_json("posts.json")
Posts.head()



,userId,id,title,body
0,1,1,sunt aut facere repellat provident occaecati e...,quia et suscipit\nsuscipit recusandae consequu...
1,1,2,qui est esse,est rerum tempore vitae\nsequi sint nihil repr...
2,1,3,ea molestias quasi exercitationem repellat qui...,et iusto sed quo iure\nvoluptatem occaecati om...
3,1,4,eum et est occaecati,ullam et saepe reiciendis voluptatem adipisci\...
4,1,5,nesciunt quas odio,repudiandae veniam quaerat sunt sed\nalias aut...
